### DISCUSSION TITLE
Does typst support knuth-Plass wrap line algorithm?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [typst/typst#626] [Q&A] [answered]
<https://en.wikipedia.org/wiki/Line_wrap_and_word_wrap>
<https://github.com/bramstein/typeset>

**Comment by laurmaedje** [comment, marked as answer]
Yes, it happens automatically if you enable justification with `#set par(justify: true)`. You can also enable it separately from justification with `#set par(linebreaks: "optimized")`.
